CRM/Core/BAO/File.php

```diff
@@ -766,23 +764,23 @@ public static function generateFileHash($entityId = NULL, $fileId = NULL, $genTs
     }
     // Trim 8 chars off the string, make it slightly easier to find
     // but reveals less information from the hash.
-    $cs = hash_hmac('sha256', "entity={$entityId}&file={$fileId}&life={$life}", $siteKey);
+    $cs = hash_hmac('sha256', "file={$fileId}&life={$life}", $siteKey);
```

Comment:
@totten this is the meat of the change (the rest of this PR just cleans up the unused param). I don't know why $entityId was included in the hash in the first place, maybe someone thought "more variables = more entropy = more secure" but the coupling with `$entityId` prevents us from generating links to files that don't have an entry in the `civicrm_entity_file` table, and the hash looks fine to me before&after. Ok with you?

Reply:
Yeah, I suspect that (historically) the parameters were defined before the hash. (If you didn't have the hash, then you might require the tuple as mitigation against scanning.) But I agree -- since we have `$cs` in the URL, the `$entityId` looks redundant.

There is a general expectation like this:

> `Activity#123` has attachment `File#456`. To read `File#123`, you must have permission to its owner `Activity#123`.

I'm guessing that there's some other mechanism by which this expectation is satisfied?

Reply:
> I'm guessing that there's some other mechanism by which this expectation is satisfied?

@totten no, to my knowledge nothing along those lines has ever been enforced. The `validateFileHash()` function does not check entity permissions and neither does anything else in the file subsystem that I'm aware of.

Reply:
Aside: Changing the formula could expire any external links during an upgrade. But this uses `$life` / `checksum_timeout`, so they're expected to expire anyway. Doesn't feel like a big problem.

(*Well, maybe if there was some customization that put links into emails? Hard to know everything that happens. Maybe it would bad to change the formula every month. But IMHO, it's okay to tweak on a rare/needful occasion.*)